src/pineko/fonll.py

```diff
@@ -0,0 +1,274 @@
+"""Module to manage FONLL predictions."""
+
+import copy
+import json
+import tempfile
+from pathlib import Path
+
+import numpy as np
+import pineappl
+import rich
+import yaml
+
+from . import configs
+
+
+class FONLLInfo:
+    """Class containing all the information for FONLL predictions."""
+
+    def __init__(
+        self, ffns3, ffn03, ffns4, ffns4til, ffns4bar, ffn04, ffns5, ffns5til, ffns5bar
+    ) -> None:
+        """Initialize fonll info."""
+        self.ffns3 = ffns3
+        self.ffn03 = ffn03
+        self.ffns4 = ffns4
+        self.ffns4til = ffns4til
+        self.ffns4bar = ffns4bar
+        self.ffn04 = ffn04
+        self.ffns5 = ffns5
+        self.ffns5til = ffns5til
+        self.ffns5bar = ffns5bar
+
+    @property
+    def fk_paths(self):
+        """Returns the list of the fk paths needed to produce FONLL predictions."""
+        paths = {
+            "ffns3": self.ffns3,
+            "ffn03": self.ffn03,
+            "ffns4": self.ffns4,
+            "ffns4til": self.ffns4til,
+            "ffns4bar": self.ffns4bar,
+            "ffn04": self.ffn04,
+            "ffns5": self.ffns5,
+            "ffns5til": self.ffns5til,
+            "ffns5bar": self.ffns5bar,
+        }
+        return {p: Path(paths[p]) for p in paths if paths[p] is not None}
```

Comment:
In the case of a list of tuples above, you do not need the whole spell out, but just to index with two indices, e.g. `self.elems[nf, M0]` (with `M0 = 3`)